specs/altair/light-client/p2p-interface.md

```diff
@@ -0,0 +1,56 @@
+# Ethereum Altair Light Client P2P Interface
+
+**Notice**: This document is a work-in-progress for researchers and implementers.
+
+This document contains the networking specification for [minimal light client](./sync-protocol.md).
+This document should be viewed as a patch to the [Altair networking specification](../altair/p2p-interface.md).
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [The Req/Resp domain](#the-reqresp-domain)
+  - [Messages](#messages)
+    - [GetLightClientSnapshot](#getlightclientsnapshot)
+    - [LightClientUpdate](#lightclientupdate)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## The Req/Resp domain
+
+### Messages
+
+#### GetLightClientSnapshot
+
+**Protocol ID:** `/eth2/beacon_chain/req/get_light_client_snapshot/1/`
+
+No Request Content.
+
+Response Content:
+
+```
+(
+  GetLightClientSnapshot
+)
+```
+
+The `GetLightClientSnapshot` SSZ container defined in [light client sync protocol](./sync-protocol.md#lightclientsnapshot).
+
+#### LightClientUpdate
+
+**Protocol ID:** `/eth2/beacon_chain/req/light_client_update/1/`
+
+Request Content:
+
+```
+(
+  LightClientUpdate
+)
+```
+
+No Response Content.
```

Comment:
If this is a push-update, and there is no explicit subscription kind of approach for a light-client, then we should spec out when and how often the server pushes the update.

Reply:
Relying on LightClientUpdate data structure to update a lightclient head is very wasteful since it only need to get the next_sync_committee pubkeys once every 256 epochs. To be honest LightClientUpdate works fine in a spec but it's not great in practice. The process of syncing can be split into two steps:
- Get the pubkeys of the current sync committee. Use req/resp on demand to query LightClientUpdate(s)
- Get latest SyncAggregate + header. More optimal to be gossiped once or more per slot.

However, gossiping SyncAggregate has an issue of how to decide which messages to accept and ignore. There are many combinations of the same signatures that results in different SyncAggregate objects (i.e. https://github.com/ethereum/consensus-specs/issues/2183).